src/py/flwr/common/record/parametersrecord.py

```diff
@@ -117,3 +117,17 @@ def __init__(
                 self[k] = array_dict[k]
                 if not keep_input:
                     del array_dict[k]
+
+    def count_bytes(self) -> int:
+        """Return number of Bytes stored in this object."""
+        num_bytes = 0
+
+        for v in self.values():
+            if v.dtype and v.shape:
+                num_bytes += int(np.prod(v.shape)) * np.dtype(v.dtype).itemsize
+            else:
+                # If datatype and shape aren't encoding during serialization
+                # we need to deserialize the data to accurately count its
+                # footprint in Bytes.
+                num_bytes += v.numpy().nbytes
```

Comment:
One potential issue is that `numpy()` method does not always convert an `Array` to `NDArray`. It can only complete the conversion when `stype` is `SType.NUMPY` (which indicates the `data` contains serialized ndarrays).

Reply:
done. we're counting now the size of the buffer array